src/GafferSceneUI/LightPositionTool.cpp

```diff
@@ -0,0 +1,347 @@
+//        copyright notice, this list of conditions and the following
+//        disclaimer.
+//
+//      * Redistributions in binary form must reproduce the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer in the documentation and/or other materials provided with
+//        the distribution.
+//
+//      * Neither the name of John Haddon nor the names of
+//        any other contributors to this software may be used to endorse or
+//        promote products derived from this software without specific prior
+//        written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#include "GafferSceneUI/LightPositionTool.h"
+#include "GafferSceneUI/SceneView.h"
+
+#include "GafferSceneUI/ContextAlgo.h"
+
+#include "Gaffer/MetadataAlgo.h"
+
+#include "IECore/AngleConversion.h"
+
+IECORE_PUSH_DEFAULT_VISIBILITY
+#include "OpenEXR/OpenEXRConfig.h"
+#if OPENEXR_VERSION_MAJOR < 3
+#include "OpenEXR/ImathEuler.h"
+#include "OpenEXR/ImathMatrixAlgo.h"
+#else
+#include "Imath/ImathEuler.h"
+#include "Imath/ImathMatrixAlgo.h"
+#endif
+IECORE_POP_DEFAULT_VISIBILITY
+
+#include "boost/algorithm/string/predicate.hpp"
+#include "boost/bind/bind.hpp"
+
+#include "fmt/format.h"
+
+using namespace boost::placeholders;
+using namespace Imath;
+using namespace Gaffer;
+using namespace GafferUI;
+using namespace GafferScene;
+using namespace GafferSceneUI;
+
+GAFFER_NODE_DEFINE_TYPE( LightPositionTool );
+
+LightPositionTool::ToolDescription<LightPositionTool, SceneView> LightPositionTool::g_toolDescription;
+size_t LightPositionTool::g_firstPlugIndex = 0;
+
+LightPositionTool::LightPositionTool( SceneView *view, const std::string &name ) :
+	TransformTool( view, name ),
+	m_shadowPivot( std::nullopt ),
+	m_shadowPoint( std::nullopt )
+{
+	SceneGadget *sg = runTimeCast<SceneGadget>( this->view()->viewportGadget()->getPrimaryChild() );
+	// We have to insert this before the underlying SelectionTool connections or it starts an object drag.
+	sg->buttonPressSignal().connectFront( boost::bind( &LightPositionTool::buttonPress, this, ::_2 ) );
```

Comment:
It'd be nice to support drags for the shadow placement - having placed the pivot already, it would be great to get continuous feedback from a drag rather than making a series of clicks. If we did that, it seems like we might as well support the same for the pivot as well?